ProcessesApi/V1/Gateways/ProcessesGateway.cs

```diff
@@ -33,47 +31,14 @@ public async Task<Process> GetProcessById(Guid id)
         }
 
         [LogCall]
-        public async Task<Process> CreateNewProcess(CreateProcessQuery query, string processName)
+        public async Task<Process> SaveProcess(Process query)
         {
-            _logger.LogDebug("Calling IDynamoDBContext.SaveAsync");
+            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync for id {query.Id}");
             var processDbEntity = query.ToDatabase();
-            processDbEntity.ProcessName = processName;
 
             await _dynamoDbContext.SaveAsync(processDbEntity).ConfigureAwait(false);
             return processDbEntity.ToDomain();
         }
 
-        private ProcessesDb CreateUpdatedProcess(ProcessesDb process, UpdateProcessQueryObject requestObject, UpdateProcessQuery query)
-        {
-            process.PreviousStates.Add(process.CurrentState);
-            process.CurrentState = new ProcessState
-            {
-                StateName = query.ProcessTrigger,
-                ProcessData = new ProcessData
-                {
-                    FormData = requestObject.FormData,
-                    Documents = requestObject.Documents
-                },
-                CreatedAt = DateTime.UtcNow,
-                UpdatedAt = DateTime.UtcNow
```

Comment:
Where do these 2 properties get set for both Create and Update scenarios?

Reply:
They are both set [here](https://github.com/LBHackney-IT/processes-api/blob/MTTL-1969/ProcessesApi/V1/UseCase/SoleToJointService.cs#L44)

Reply:
That works for now.
It will have to get updated in the next story as the Updated value will need to get (re)set in the Gateway save method.
When you have created/updated timestamps on db objects it's generally good practice to set them as close as possible to the call to save the record.

Reply:
Will Keep that in mind